### PR TITLE
ogr2ogr: error out if WKT argument of -clipsrc/-clipdst is an invalid geometry

### LINKED ISSUE #10289: ogr2ogr -clipsrc/-clipdst: TopologyException: side location conflict: perhaps should be fatal
### Feature description

It seems this should be a fatal error. The output file made was a mess, at least in my case.
```
$ ogr2ogr m.kmz l.kmz -clipsrc 'POLYGON ((120.79382 24.20545 ...
ERROR 1: TopologyException: side location conflict at
  120.7946631515733 24.205405693211443.
  This can occur if the input geometry is invalid.
$ echo $?
0
```
To reproduce use some POLYGON that is really a figure 8, etc.
Version 3.9.

### PR DESCRIPTION
**Pull request by rouault**
If specifying -makevalid, MakeValid() will be run on it to make it valid.

For consistency, change behavior of -clipsrc/-clipdst {datasetname} to not automatically make valid invalid clip geometries, but do it only if -makevalid is specified.

Fixes #10289

### CODE REVIEW

File: apps/ogr2ogr_lib.cpp [hunk 1]
```diff
@@ -2328,19 +2340,58 @@ GDALDatasetH GDALVectorTranslate(const char *pszDest, GDALDatasetH hDstDS,
             *pbUsageError = TRUE;
         return nullptr;
     }
+    if (psOptions->poClipSrc && !psOptions->poClipSrc->IsValid())
+    {
+        if (!psOptions->bMakeValid)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                     "-clipsrc geometry is invalid. You can try to make it "
+                     "valid with -makevalid");
```
Comment (dbaston) [inline apps/ogr2ogr_lib.cpp:2349]: The results of MakeValid can be surprising.

Do you think it makes sense to either extend this message along the lines of

> -clipsrc geometry is invalid. You can try to make it valid with -makevalid. The results should be manually inspected.

or instead have a separate warning when MakeValid is actually used?

> The -clipsrc geometry is invalid and was corrected with MakeValid. The results of this operation should be inspected.

Comment (rouault) [reply apps/ogr2ogr_lib.cpp:2349]: > The results should be manually inspected.

Not sure if that would be very helpful, as the user has no easy way to know how the geometry was corrected by MakeValid()

Comment (dbaston) [reply apps/ogr2ogr_lib.cpp:2349]: Maybe more clear to say "the output of ogr2ogr should be manually inspected."

Comment (rouault) [reply apps/ogr2ogr_lib.cpp]: ok, warning/error messages tuned

**Review by dbaston — APPROVED** [review]

**Pull request merged**